pyproject.toml

```diff
@@ -165,6 +167,7 @@ dev = [
     "mkdocs-material>=9.6.4",
     "mkdocstrings[python]>=0.28.0",
     "mkdocs-gen-files>=0.5.0",
+    "whenever>=0.7.3,<1.0.0",
```

Comment:
Should this be python conditional, or are we using it in a non-version-specific way?  (Skimming PR from top to bottom, ignore if this answer becomes clear at line 3,215 :P)

Reply:
We want to install it in all versions for the dev dependency group because we're using `whenever` in some tests where `pendulum` was necessary.